### PR TITLE
Use go standard library for common bit operations

### PR DESCRIPTION
**Pull request by kad**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
PR #72913 introduced own versions of the bit operations that are less efficient than ones from standard library. Let's not re-invent the wheel and use ones from standard library.

**Special notes for your reviewer**:
cc: @tedyu @dashpole  @klueska
PR #78451 will require rebase after this one.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
Since XOr is not accepted yet, can you remove it from your PR ?

I will rebase mine once this PR gets merged.

Thanks

**Comment by kad** [comment]
> Since XOr is not accepted yet, can you remove it from your PR ?
> I will rebase mine once this PR gets merged.

Done.

**Review by klueska — APPROVED** [review]

File: pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go
Comment (mattjmcnaughton) [inline pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go]: I'm having a little trouble understanding how the previous version of this test passed? It seems like if the mask bit is 0, then there should be no 1's right? similarly, if the mask bit is 42, shouldn't the count of 1s be more than 1?

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
Thanks for thinking about this - I agree, its good to use the std lib whenever possible :)

I am _far_ from an expert on bit operations, so apologies if these are goofy questions:

1. Is it not a problem that the new `String` method has a different output than the old `String` method? Could this cause incorrect behavior or be confusing to users?

2. See my question on test :) I think we could improve readability by having a brief comment on this test with the answer to my question.

Thanks again !

File: pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go
Comment (klueska) [reply pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go]: The previous test ssets bit 0 in the mask, so 1 bit is set. The new test just changes this to bit 42. There is no real reason to pick bit 42 over bit 0. It doesn't change the outcome.

**Comment by klueska** [comment]
@mattjmcnaughton, Regarding:

> Is it not a problem that the new String method has a different output than the old String method? Could this cause incorrect behavior or be confusing to users?

Please see:
https://github.com/kubernetes/kubernetes/pull/72913#discussion_r288200742

File: pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go
Comment (kad) [reply pkg/kubelet/cm/topologymanager/socketmask/socketmask_test.go]: @mattjmcnaughton previous version of the test sets first bit. For function that supposed to count amount of 1s in the bitmask, is a very trivial case. It would be of course better to have test completely rewritten to have it tested 2-3 bits set in the beginning/middle/end of the range... But well, I think checking that mask has 1 bit set somewhere in the middle is good enough for updated code that uses standard library.

**Comment by kad** [comment]
> 1. Is it not a problem that the new `String` method has a different output than the old `String` method? Could this cause incorrect behavior or be confusing to users?

`String()` so far used only in few tests and in some log messages. It shouldn't be used for any real operations with the value. For people who check logs, it is good to use standard representation of binary value rather than something else.

**Review by mattjmcnaughton — APPROVED** [review]
thanks for the additional info :)

/lgtm

**Comment by klueska** [comment]
This PR appears to be blocked because it thinks it needs the v1.15 milestone on it. It should be OK to merge at this point without it.

**Comment by nolancon** [comment]
Can we get approval on this PR @mattjmcnaughton @sjenning @dashpole

**Comment by mattjmcnaughton** [comment]
/assign @dchen1107 @derekwaynecarr 

@nolancon in the future, using "/ assign" to assign the suggested approvers is one of the quickest ways to get a diff approved :)

File: pkg/kubelet/cm/topologymanager/socketmask/socketmask.go
Comment (ConnorDoyle) [inline pkg/kubelet/cm/topologymanager/socketmask/socketmask.go]: 💯

**Review by ConnorDoyle — APPROVED** [review]
/approve
/lgtm

Thanks @kad

**Pull request merged**
